src/wp-includes/blocks/index.php

```diff
@@ -12,6 +12,77 @@
 require BLOCKS_PATH . 'widget-group.php';
 require BLOCKS_PATH . 'require-dynamic-blocks.php';
 
+/**
+ * Registers core block style handles.
+ *
+ * While {@see register_block_style_handle()} is typically used for that, the way it is
+ * implemented is inefficient for core block styles. Registering those style handles here
+ * avoids unnecessary logic and filesystem lookups in the other function.
+ *
+ * @since 6.3.0
+ */
+function register_core_block_style_handles() {
+	if ( ! wp_should_load_separate_core_block_assets() ) {
+		return;
+	}
+
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require ABSPATH . WPINC . '/blocks/blocks-json.php';
+	}
+
+	$includes_url   = includes_url();
+	$suffix         = wp_scripts_get_suffix();
+	$wp_styles      = wp_styles();
+	$style_fields   = array(
+		'style',
+		'editorStyle',
+	);
+	$transient_name = 'block_styles';
```

Comment:
Since the styles are all from core itself, I think we can easily handle invalidation here when WordPress is updated: Either we add `$wp_version` to the transient name, or we delete the transient on every WordPress update (like you're doing above). In order to support programmatically controlled sites (e.g. that update via Composer), it may be more reliable to rely on the `$wp_version` global.

Also, let's prefix this with something like `wp_core_`, to clarify that it is only for WP core block styles.
```suggestion
	$transient_name = 'wp_core_block_styles_' . $GLOBALS['wp_version'];
```

Reply:
@felixarntz I originally had it like this. But as the tranient would be stored in options. This would mean that a version of the transient would be stored in an autoloaded option forever more right?

Reply:
@spacedmonkey Fair point, in that case, let's just change to 
```suggestion
	$transient_name = 'wp_core_block_styles';
```